pkg/services/syncer_custom.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	upgradev1 "github.com/rancher/system-upgrade-controller/pkg/apis/upgrade.cattle.io/v1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	provv1 "github.com/rancher-sandbox/rancheros-operator/pkg/apis/rancheros.cattle.io/v1"
+	"github.com/rancher-sandbox/rancheros-operator/pkg/clients"
+)
+
+type CustomSyncer struct {
+	upgradev1.ContainerSpec
+}
+
+func (j *CustomSyncer) toContainers() []corev1.Container {
+	return []corev1.Container{
+		{
+			VolumeMounts: []corev1.VolumeMount{{Name: "output",
+				MountPath: "/output",
+			}},
+			Name:    "runner",
+			Image:   j.Image,
+			Command: j.Command,
+			Args:    j.Args,
+			EnvFrom: j.EnvFrom,
+			Env:     j.Env,
+		},
+	}
+}
+
+func (j *CustomSyncer) sync(s provv1.ManagedOSVersionChannel, c *clients.Clients) ([]provv1.ManagedOSVersion, error) {
+	logrus.Infof("Syncing '%s/%s'", s.Namespace, s.Name)
+
+	serviceAccount := false
+	p, err := c.Core.Pod().Get(s.Namespace, s.Name, v1.GetOptions{})
+	if err != nil {
+		_, err = c.Core.Pod().Create(&corev1.Pod{
+			TypeMeta: v1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      s.Name,
+				Namespace: s.Namespace,
+				OwnerReferences: []v1.OwnerReference{
+					*v1.NewControllerRef(&s, provv1.SchemeGroupVersion.WithKind("ManagedOSVersionChannel")),
```

Comment:
According to my experience on #20 the created reference is blocking the deletion by default, that's why I manually assigned `ref.BlockOwnerDeletion = false` in #20. To me it is a far from intuitive default but it is also documented on k8s [docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/) 🤷🏽‍♂️:

> Kubernetes automatically sets this field to true if a [controller](https://kubernetes.io/docs/concepts/architecture/controller/) (for example, the Deployment controller) sets the value of the metadata.ownerReferences field. You can also set the value of the blockOwnerDeletion field manually to control which dependents block garbage collection.

Reply:
I think in this case we want it to true, no? we don't want the owner to disappear due to gc meanwhile we are running a pod sync